pkg/controllers/common/utils.go

```diff
@@ -106,3 +107,14 @@ func NodeIsMaster(node *v1.Node) bool {
 	}
 	return false
 }
+
+// NumReconcile depends on the https://github.com/uber-go/automaxprocs which will set the number of CPUs from cgroups information.
+// automaxprocs only support linxu platform.
+func NumReconcile() int {
+	numCPU := runtime.GOMAXPROCS(0)
+	if numCPU <= 1 {
```

Comment:
Why hard code with 4 or 8 instead of return NumCPU or GOMAXPROCS?

Reply:
nsx-operator is a network-intensive task， so each CPU should have more than 1 thread to utilize CPU resource. 4 refers to NCP setting